ddtrace/tracer/sqlcomment.go

```diff
@@ -0,0 +1,174 @@
+	c = new(SQLCommentCarrier)
+	c.Mode = mode
+	c.Query = query
+	c.SpanID = random.Uint64()
+	return c
+}
+
+// Inject injects a span context in the carrier's query.
+func (c *SQLCommentCarrier) Inject(spanCtx ddtrace.SpanContext) error {
+	if c.Mode == CommentInjectionDisabled {
+		return nil
+	}
+
+	tags := make(map[string]string)
+	if c.Mode == ServiceTagsInjection || c.Mode == FullSQLCommentInjection {
+		ctx, ok := spanCtx.(*spanContext)
+		var env, version string
+		if ok {
+			if e, ok := ctx.meta(ext.Environment); ok {
+				env = e
+			}
+			if v, ok := ctx.meta(ext.Version); ok {
+				version = v
+			}
+		}
+		if globalconfig.ServiceName() != "" {
+			tags[ServiceNameSQLCommentKey] = globalconfig.ServiceName()
+		}
+		if env != "" {
+			tags[ServiceEnvironmentSQLCommentKey] = env
+		}
+		if version != "" {
+			tags[ServiceVersionSQLCommentKey] = version
+		}
+	}
+	if c.Mode == FullSQLCommentInjection {
+		samplingPriority := 0
+
+		var traceID uint64
+		ctx, ok := spanCtx.(*spanContext)
+		if ok {
+			if sp, ok := ctx.samplingPriority(); ok {
+				samplingPriority = sp
+			}
+			if ctx.TraceID() > 0 {
+				traceID = ctx.TraceID()
+			}
+		}
+		if traceID == 0 {
+			traceID = c.SpanID
+		}
+		tags[TraceIDSQLCommentKey] = strconv.FormatUint(traceID, 10)
+		tags[SpanIDSQLCommentKey] = strconv.FormatUint(c.SpanID, 10)
+		tags[SamplingPrioritySQLCommentKey] = strconv.Itoa(samplingPriority)
+	}
+
+	c.Query = commentQuery(c.Query, tags)
+	return nil
+}
+
+// commentQuery returns the given query with the tags from the SQLCommentCarrier applied to it as a
+// prepended SQL comment. The format of the comment follows the sqlcommenter spec.
+// See https://google.github.io/sqlcommenter/spec/ for more details.
+func commentQuery(query string, tags map[string]string) string {
+	c := serializeTags(tags)
+	if c == "" {
+		return query
+	}
+	if query == "" {
+		return c
+	}
+	return fmt.Sprintf("%s %s", c, query)
+}
+
+func serializeTags(tags map[string]string) (comment string) {
+	if len(tags) == 0 {
+		return ""
+	}
+	serializedTags := make([]string, 0, len(tags))
+	for k, v := range tags {
+		serializedTags = append(serializedTags, serializeTag(k, v))
+	}
+	sort.Strings(serializedTags)
+	comment = strings.Join(serializedTags, ",")
+	return fmt.Sprintf("/*%s*/", comment)
+}
+
+func serializeTag(key string, value string) (serialized string) {
+	sKey := serializeKey(key)
+	sValue := serializeValue(value)
+
+	return fmt.Sprintf("%s=%s", sKey, sValue)
+}
+
+func serializeKey(key string) (encoded string) {
+	enc := urlEncode(key)
+	return escapeMetaChars(enc)
+}
+
+func serializeValue(val string) (encoded string) {
+	enc := urlEncode(val)
+	escapedMeta := escapeMetaChars(enc)
+	return escapeSQL(escapedMeta)
+}
+
+func urlEncode(val string) string {
+	e := url.QueryEscape(val)
+	return strings.Replace(e, "+", "%20", -1)
+}
+
+func escapeSQL(value string) (escaped string) {
+	return fmt.Sprintf("'%s'", value)
+}
```

Comment:
I really don't think we need this function just for one line of code used in just one place.